### PR TITLE
Fix out-of-bound checking in BidirectionalEndpointSpanExtractor for empty sequences.

### PR DESCRIPTION
**Pull request by wammar**

### CODE REVIEW

**Comment by brendan-ai2** [comment]
Thanks for the fix! Ideally we'd add a test to https://github.com/allenai/allennlp/blob/master/allennlp/tests/modules/span_extractors/bidirectional_endpoint_span_extractor_test.py. In the short term, could you at least add:

```
@pytest.mark.skip(reason="TODO(brendanr/waleeda): Flesh this out.")
def test_handles_empty_sequences(self):
    pass
```

**Comment by wammar** [comment]
Thanks for the review, Brendan! I added `test_forward_doesnt_raise_with_empty_sequence` which demonstrates why this change is necessary.

File: allennlp/tests/modules/span_extractors/bidirectional_endpoint_span_extractor_test.py [hunk 1]
```diff
@@ -172,6 +172,25 @@ def test_correct_sequence_elements_are_embedded_with_a_masked_sequence(self):
         numpy.testing.assert_array_equal(backward_start_embeddings.data.numpy(),
                                          correct_backward_start_embeddings.data.numpy())
 
+
+    def test_forward_doesnt_raise_with_empty_sequence(self):
+        # size: (batch_size=1, sequence_length=1, emb_dim=2)
```
Comment (brendan-ai2) [inline allennlp/tests/modules/span_extractors/bidirectional_endpoint_span_extractor_test.py]: Looks like `sequence_length=2`, right?

File: allennlp/tests/modules/span_extractors/bidirectional_endpoint_span_extractor_test.py [hunk 2]
```diff
@@ -172,6 +172,25 @@ def test_correct_sequence_elements_are_embedded_with_a_masked_sequence(self):
...
+        sequence_tensor = torch.FloatTensor([[[0., 0.], [0., 0.]]])
+        # size: (batch_size=1, sequence_length=1)
+        sequence_mask = torch.LongTensor([[0]])
```
Comment (brendan-ai2) [inline allennlp/tests/modules/span_extractors/bidirectional_endpoint_span_extractor_test.py]: The `sequence_length` here should match the `sequence_tensor`. So:

```
sequence_mask = torch.LongTensor([[0, 0]])
```

IIUC.

**Comment by wammar** [comment]
Good catch, thank you! Now fixed.

**Comment by brendan-ai2** [comment]
I'm going to go ahead and merge this. The fix is needed to fully validate the `Scorer` refactor.

**Pull request merged**